src/main/java/tools/jackson/databind/deser/impl/MethodProperty.java

```diff
@@ -196,11 +192,30 @@ public Object setAndReturn(DeserializationContext ctxt,
             }
         }
         try {
-            Object result = _setter.invoke(instance, value);
+            Object result = _setterReturn.get().invokeExact(instance, value);
             return (result == null) ? instance : result;
-        } catch (Exception e) {
+        } catch (Throwable e) {
             _throwAsJacksonE(ctxt.getParser(), e, value);
             return null;
         }
     }
+
+    class SetterHolder extends UnreflectHandleSupplier {
+        private static final long serialVersionUID = 1L;
```

Comment:
Almost certainly no longer needed (I removed most `implements java.io.Serializable` cases in #5143)

Reply:
@cowtowncoder , thank you for suggesting this, unfortunately I do not think it can be removed entirely. This is because we would like to initialize these `MethodHandle` instances in the constructor, however we cannot do so because `fixAccess` is an instance method. So we must defer init until after `fixAccess` might be called. Unfortunately, there is no guarantee `fixAccess` is called at all - it might be disabled - leaving no natural place to actually initialize the MethodHandle.

So, I think lazily setting it up once at time of use is the least bad path for now. This is why I think it would be nice if the access-fixing could happen at constructor time, rather than a virtual method call.

I pushed an update to remove `Serializable`, at least, since that is no longer needed.

Reply:
Oh, I see, you are literally referring to the `implements Serializable`. So yes, I removed that :smile: